src/iop/basicadj.c

```diff
@@ -0,0 +1,1744 @@
+/*
+    This file is part of darktable,
+    copyright (c) 2019 edgardo hoszowski.
+
+    darktable is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    darktable is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/*
+    auto exposure is based on RawTherapee's Auto Levels
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "bauhaus/bauhaus.h"
+#include "develop/imageop.h"
+#include "gui/color_picker_proxy.h"
+
+DT_MODULE_INTROSPECTION(1, dt_iop_basicadj_params_t)
+
+#define exposure2white(x) exp2f(-(x))
+
+typedef enum dt_iop_basicadj_preservecolors_t {
```

Comment:
{ should be on new line

Reply:
We still have this clang issue, even if I change it, clang will change it back.